app/viewmodels/package.scala

```diff
@@ -695,24 +697,50 @@ package object viewmodels {
     }
   }
 
+  implicit class RichRichGuaranteeTypeSeq(value: Seq[generated.GuaranteeType03]) {
+
+    def normaliseOtherGuaranteeRefs: Seq[generated.GuaranteeType03] =
+      value
+        .groupBy(_.sequenceNumber)
+        .map {
+          (_, list) =>
+            val combinedOtherGuaranteeRefs = {
+              val refs = list.flatMap(_.otherGuaranteeReference)
+              if (refs.isEmpty) None
+              else Some(refs.takeSampleWithoutPeriod)
+            }
+            list.head.copy(otherGuaranteeReference = combinedOtherGuaranteeRefs)
+        }
+        .toSeq
+
+  }
+
   implicit class RichGuaranteeType03(value: GuaranteeType03) {
 
-    def asString: String = Seq(
-      Some(value.sequenceNumber.toString),
-      Some(value.guaranteeType),
-      Some(value.GuaranteeReference.map(_.asString).take2(_.semiColonSeparate)),
-      value.otherGuaranteeReference
-    ).flatten.commaSeparate
+    def asString: String = value.guaranteeType match {
+      case "5" | "A" =>
+        Seq(
+          value.sequenceNumber.toString,
+          value.guaranteeType
+        ).slashSeparate
+      case "0" | "1" | "2" | "4" | "9" =>
+        Seq(
+          value.sequenceNumber.toString,
+          value.guaranteeType,
+          value.GuaranteeReference.flatMap(_.asString).takeSampleWithoutPeriod
+        ).slashSeparate
+      case "3" | "8" =>
+        Seq(
+          Some(value.sequenceNumber.toString),
+          Some(value.guaranteeType),
+          value.otherGuaranteeReference
+        ).flatten.slashSeparate
+    }
```

Comment:
We probably don't need to worry about these cases. We wouldn't be getting to the point of them receiving an IE029 if their IE015 didn't have valid guarantee data. We can just do a one-size-fits-all here.

Reply:
Just jumping on this, can you explain further @jackheslop96, I thought the case approach is to cater for the Guarantee reference field bit that might be present depending on the "type"?

Reply:
@justega247 The rules in the DDNTA mean that certain data items will only be present for certain guarantee types. You can see this in our guarantee details front end navigation logic. So in production, by virtue of receiving a valid IE029, these cases become obsolete. If that makes sense?

We can review the rules tomorrow to see if I’ve got the wrong end of the stick!